cm2350/e200_gdb.py

```diff
@@ -86,6 +85,13 @@ def getTargetXml(self, reggrps):
         self._genRegPktFmt()
         self._updateEnviGdbIdxMap()
 
+        # To mimic what the GdbBaseEmuServer.getTargetXml() function does, go 
+        # find the gdb register indexes for the halt
+        for reg_name, (_, reg_idx) in self._gdb_reg_fmt.items():
+            envi_idx = self._gdb_to_envi_map[reg_idx][vtp_gdb.GDB_TO_ENVI_IDX]
+            if envi_idx in haltregs or reg_name in haltregs:
+                self._haltregs.append(reg_idx)
```

Comment:
What does this do?

Reply:
This was to get the halt registers populated based on the register names, these registers are sent to the GDB client when the target stops. this lookup list helps the gdbstub identify which registers to send.

Reply:
wait, what?  Registers are sent to GDB Client on halt?  That's cool, I didn't realize that.